src/main/java/de/tum/in/www1/artemis/web/rest/hestia/ExerciseHintResource.java

```diff
@@ -105,44 +135,61 @@ public ExerciseHintResource(ExerciseHintRepository exerciseHintRepository, Autho
         authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.EDITOR, hintBeforeSaving.getExercise(), null);
         authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.EDITOR, exerciseHint.getExercise(), null);
         ExerciseHint result = exerciseHintRepository.save(exerciseHint);
-        return ResponseEntity.ok().headers(HeaderUtil.createEntityUpdateAlert(applicationName, true, ENTITY_NAME, exerciseHint.getId().toString())).body(result);
+        return ResponseEntity.ok().headers(HeaderUtil.createEntityUpdateAlert(applicationName, true, EXERCISE_HINT_ENTITY_NAME, exerciseHint.getId().toString())).body(result);
     }
 
     /**
-     * GET /exercise-hints/:hintId/title : Returns the title of the hint with the given id
+     * GET exercises/:exerciseId/exercise-hints/:hintId/title : Returns the title of the hint with the given id
      *
-     * @param hintId the id of the hint
+     * @param exerciseHintId the id of the exerciseHint
+     * @param exerciseId the exerciseId of the exercise of which to retrieve the exerciseHints' title
      * @return the title of the hint wrapped in an ResponseEntity or 404 Not Found if no hint with that id exists
+     * or with status {@code 409 (Conflict)} if the exerciseId is not valid.
      */
-    @GetMapping(value = "/exercise-hints/{hintId}/title")
+    @GetMapping("exercises/{exerciseId}/exercise-hints/{exerciseHintId}/title")
     @PreAuthorize("hasRole('USER')")
-    public ResponseEntity<String> getHintTitle(@PathVariable Long hintId) {
-        final var title = exerciseHintRepository.getHintTitle(hintId);
-        return title == null ? ResponseEntity.notFound().build() : ResponseEntity.ok(title);
+    public ResponseEntity<String> getHintTitle(@PathVariable Long exerciseId, @PathVariable Long exerciseHintId) {
```

Comment:
This endpoint is called in quite a few places on the client. I don't know the reasoning behind this, so I don't know if we should change this in this PR.

Reply:
I see. This would not be within the scope of this PR.
Please consider though if it's within the scope of your thesis. As of now, I think this would be a nice code improvement for another PR.

Reply:
We will most likely come back to this at some point in our thesis, as we will also need to change how hints are displayed in the client, so that would be the best point to remove this endpoint.